homeassistant/components/device_tracker/__init__.py

```diff
@@ -660,6 +660,17 @@ def async_get_device_name(self, device: str) -> Any:
         """
         return self.hass.async_add_job(self.get_device_name, device)
 
+    def get_extra_attributes(self, device: str) -> dict:
+        """Get the extra attributes of a device."""
+        raise NotImplementedError()
+
+    def async_get_extra_attributes(self, device: str) -> Any:
```

Comment:
This should be made a coroutine method with `async def`. It's a bad pattern to return a coroutine object from a non coroutine function.